alibi_detect/od/_svm.py

```diff
@@ -0,0 +1,233 @@
+from typing import TYPE_CHECKING, Any, Dict, Optional, Tuple, Union
+
+import numpy as np
+
+from alibi_detect.base import (BaseDetector, FitMixin, ThresholdMixin,
+                               outlier_prediction_dict)
+from alibi_detect.exceptions import _catch_error as catch_error
+from alibi_detect.od.pytorch import DgSVMTorch, SdgSVMTorch
+from alibi_detect.utils._types import Literal
+from alibi_detect.utils.frameworks import BackendValidator
+from alibi_detect.version import __version__
+
+if TYPE_CHECKING:
+    import torch
+
+
+backends = {
+    'pytorch': {
+        'sgd': SdgSVMTorch,
+        'gd': DgSVMTorch
```

Comment:
Here the class names have the `d` and `g` the wrong way round. Also could we perhaps use "batch gradient descent" (so `BgdSVMTorch`) for the Pytorch variant? I think it makes the distinction clearer as just gradient descent is general enough that it could refer to both variants.

Reply:
Agree with `BGD`.